signer-service/src/api/services/pendulum/pendulum.service.ts

```diff
@@ -25,14 +26,29 @@ export function getFundingData(
   return { fundingAccountKeypair, fundingAmountRaw };
 }
 
-export const fundEphemeralAccount = async (ephemeralAddress: string): Promise<boolean> => {
+export const fundEphemeralAccount = async (ephemeralAddress: string, requiresGlmr?: boolean): Promise<boolean> => {
   try {
     const apiData = await createPolkadotApi();
     const { fundingAccountKeypair, fundingAmountRaw } = getFundingData(apiData.ss58Format, apiData.decimals);
 
-    await apiData.api.tx.balances
-      .transferKeepAlive(ephemeralAddress, fundingAmountRaw)
-      .signAndSend(fundingAccountKeypair);
+    if (requiresGlmr) {
+      const { fundingAccountKeypair } = getFundingData(apiData.ss58Format, apiData.decimals);
+      const pendulumCurrencyId = { XCM: 13 };
+
+      const penFundingTx = apiData.api.tx.balances.transferKeepAlive(ephemeralAddress, fundingAmountRaw);
+      const glmrFundingTx = apiData.api.tx.tokens.transfer(
+        ephemeralAddress,
+        pendulumCurrencyId,
+        GLMR_FUNDING_AMOUNT_RAW,
+      );
+
+      const batch = apiData.api.tx.utility.batch([penFundingTx, glmrFundingTx]);
```

Comment:
Nit: `batchAll` will make sure that either both extrinsics work or none. Makes more sense to use this one here as it would be unfortunate if only one works for some reason. Then the recipient account would still hold some tokens that we need to recover (or not). 
```suggestion
      const batch = apiData.api.tx.utility.batchAll([penFundingTx, glmrFundingTx]);
```

Reply:
Good call.